packages/nhost-js/src/clients/graphql.ts

```diff
@@ -21,33 +21,47 @@ export class NhostGraphqlClient {
     })
   }
 
-  async request(
+  async request<T = any, V = any>(
     document: string,
-    variables?: any,
+    variables?: V,
     config?: AxiosRequestConfig
-  ): Promise<GraphqlRequestResponse> {
+  ): Promise<GraphqlRequestResponse<T>> {
     // add auth headers if any
     const headers = {
       ...config?.headers,
       ...this.generateAccessTokenHeaders()
     }
 
-    const operationName = ''
-
-    let responseData
     try {
-      const res = await this.instance.post(
+      const operationName = ''
+      const res = await this.instance.post<GraphqlResponse<T>>(
         '',
         {
           operationName: operationName || undefined,
           query: document,
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-          variables
+          ...(variables ? { variables } : {})
```

Comment:
For the sake of readability we can rather use `variables: variables || {}` here. I don't worry about having a `variables: {}` passed to the Axios POST call.